packages/toolpad-app/src/server/withReqResLogs.ts

```diff
@@ -0,0 +1,83 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+import * as _ from 'lodash-es';
+import { uuidv4 } from '../utils/uuid';
+import logger from './logger';
+
+function getLogMessageInfo(req: NextApiRequest) {
+  const { type, name } = req.body;
+  return type && name ? `${type}:${name}` : '';
+}
+
+function getBaseLogProperties(req: NextApiRequest) {
+  return {
+    type: req.body.type,
+    name: req.body.name,
+    ipAddress: req.headers['x-forwarded-for'] || req.socket?.remoteAddress,
+    host: req.headers.host,
+    userAgent: req.headers['user-agent'],
+    timestamp: new Date().getTime(),
+  };
+}
+
+type RestArgs = [
+  chunk: any,
+  encoding: BufferEncoding,
+  callback?: ((error?: Error | null) => void) | undefined,
+];
+
+function logRequestResponse(req: NextApiRequest, res: NextApiResponse, requestId: string) {
+  const oldWrite = res.write;
+  const oldEnd = res.end;
+
+  const chunks: Buffer[] = [];
+
+  res.write = (...restArgs: any[]) => {
```

Comment:
Could you use `RestArgs` type instead of `any[]`? would it allow us to avoid type-hinting with `as`? 🤔 
```ts
res.write = (...restArgs: RestArgs) => {
  chunks.push(Buffer.from(restArgs[0]));
  return oldWrite.apply(res, restArgs);
}
```

Reply:
It turns out I could just just remove the `any[]`, I just did.
About the `as` assertion, I think it's fine here as the restArgs can be an array with 1-3 elements so we would always need logic to assert each type anyway. And for this usecase it doesn't seem like this logic would ever break.